### PR TITLE
libreoffice: use boost176

### PR DESCRIPTION
**Pull request by Tatsh**
https://trac.macports.org/ticket/63136

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.4 20F71 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by reneeotten** [comment]
can this not use the `boost` PG? Also, when changing (library) dependencies you will need to increase the `revision`

**Comment by Tatsh** [comment]
> can this not use the `boost` PG? Also, when changing (library) dependencies you will need to increase the `revision`

This is done now.

**Comment by Tatsh** [comment]
I built this locally and did not need the `compiler.cxx_standard` line but the last build that got far enough failed configure because it said `clang++` cannot compile C++17. This has never been an issue before. It has built in that environment many times before.

**Comment by Tatsh** [comment]
I guess it works now.

File: office/libreoffice/Portfile [hunk 1]
```diff
@@ -77,7 +78,7 @@ depends_build-append \
 depends_lib-append  \
     port:apr \
     port:apr-util \
-    port:boost \
+    port:[boost::depends_portname] \
```
Comment (reneeotten) [inline office/libreoffice/Portfile:81]: with the `boost` PG you shouldn't include the port manually anymore (see also the `port lint` warning)

**Comment by reneeotten** [comment]
> I built this locally and did not need the `compiler.cxx_standard` line but the last build that got far enough failed configure because it said `clang++` cannot compile C++17. This has never been an issue before. It has built in that environment many times before.

I am not sure about that; it seems that this C++17 requirement with the new boost has come up in a few other places as well which isn't ideal... (CC @michaelld @mascguy @cjones051073)

File: office/libreoffice/Portfile [hunk 2]
```diff
@@ -208,6 +210,7 @@ configure.args-append  \
     --enable-readonly-installset \
     --enable-release-build \
     --enable-symbols \
+    --with-boost=[boost::install_area] \
```
Comment (mascguy) [inline office/libreoffice/Portfile:213]: I believe this is also done by the portgroup?

**Comment by michaelld** [comment]
IDK about Boost and C++17 specifically, but if the port is trying to use `std::filesystem` -- a C++17 feature -- I've had a lot of difficulty getting that to work with -any- Clang++

File: office/libreoffice/Portfile (hunk 2 above)
Comment (michaelld) [reply office/libreoffice/Portfile:213]: hmmm ... In my experience the primary issue with non-CMake configuration is that the flags can/will vary even between different ports. Broadly speaking `--with-boost=...` will work with autotools, but it's not guaranteed to do so. With CMake there are standard ways for finding, and so what was added is virtually universally correct across all ports -- I know of no exceptions, but that doesn't mean there aren't any!

**Pull request merged**